src/request/change-password/ChangePassword.js

```diff
@@ -0,0 +1,212 @@
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.DOWNLOAD_FILE} .progress-indicator`),
+            4,
+            4,
+        );
+
+        // Events
+
+        this._passwordGetter.on(PassphraseBox.Events.SUBMIT, this._unlock.bind(this));
+        this._passwordSetter.on(PassphraseSetterBox.Events.ENTERED, this._prepare.bind(this));
+        this._passwordSetter.on(PassphraseSetterBox.Events.SUBMIT, this._commitChangeAndOfferLoginFile.bind(this));
+        this._passwordSetter.on(PassphraseSetterBox.Events.NOT_EQUAL, () => this._loginFileIcon.unlock());
+    }
+
+    async run() {
+        this._passwordGetter.reset();
+        window.location.hash = ChangePassword.Pages.ENTER_PASSWORD;
+        if (TopLevelApi.getDocumentWidth() > Constants.MIN_WIDTH_FOR_AUTOFOCUS) {
+            this._passwordGetter.focus();
+        }
+    }
+
+    /**
+     * Called when the user enters his old password (or just clicks 'continue') to unlock his account.
+     * @param {string} oldPassword
+     */
+    async _unlock(oldPassword) {
+        TopLevelApi.setLoading(true);
+        const oldPasswordBytes = oldPassword ? Utf8Tools.stringToUtf8ByteArray(oldPassword) : undefined;
+        /** @type {Key?} */
+        let key = null;
+        try {
+            key = await KeyStore.instance.get(this._request.keyInfo.id, oldPasswordBytes);
+        } catch (e) {
+            if (e.message === 'Invalid key') {
+                TopLevelApi.setLoading(false);
+                this._passwordGetter.onPassphraseIncorrect();
+                return;
+            }
+            this._reject(new Errors.CoreError(e));
+            return;
+        }
+        if (!key) {
+            this._reject(new Errors.KeyNotFoundError());
+            return;
+        }
+        this._key = key;
+        this._passwordSetter.reset();
+        window.location.hash = ChangePassword.Pages.SET_PASSWORD;
+        if (TopLevelApi.getDocumentWidth() > Constants.MIN_WIDTH_FOR_AUTOFOCUS) {
+            this._passwordSetter.focus();
+        }
+        TopLevelApi.setLoading(false);
+    }
+
+
+    /**
+     * Called after new password was entered first time.
+     */
+    async _prepare() {
+        let colorClass = '';
+        if (this.key.secret instanceof Nimiq.Entropy) {
+            const color = Iqons.getBackgroundColorIndex(
+                new Nimiq.Address(
+                    // use color of first address as loginFile color
+                    this.key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+                ).toUserFriendlyAddress(),
+            );
+            const colorString = LoginFile.CONFIG[color].name;
+            colorClass = `nq-${colorString}-bg`;
+        }
+        this._loginFileIcon.lock(colorClass);
+        this._progressIndicator.setStep(3);
+    }
+
+    /**
+     * Called after new password was entered second time.
+     * @param {string} newPassword
```

Comment:
In case we want to allow for no password as new password this becomes 
```suggestion
     * @param {string} [newPassword]
```
That would also mean, that the test for 
```
if (this.key.secret instanceof Nimiq.PrivateKey) {
```
is no longer enough as unencrypted Entropies are also exempt from Login Files.